src/envoy/http/service_control/handler_impl.cc

```diff
@@ -143,11 +145,20 @@ void ServiceControlHandlerImpl::prepareReportRequest(
     ::espv2::api_proxy::service_control::ReportRequestInfo& info) {
   fillOperationInfo(info);
 
-  // Report: not to send api-key if invalid or service is not enabled.
-  if (check_response_info_.error_type == ScResponseErrorType::API_KEY_INVALID ||
-      check_response_info_.error_type ==
-          ScResponseErrorType::SERVICE_NOT_ACTIVATED) {
-    info.api_key.clear();
+  // This logic determines if consumer metric is sent in report based on the
+  // validity of the API Key.
+  if (on_check_done_called_) {
```

Comment:
Did not see where on_check_done_called_ is set to true.

Reply:
It is pre-existing code: https://github.com/GoogleCloudPlatform/esp-v2/blob/ec6c83737213c4ee8401c402a3e0f2984df577ba/src/envoy/http/service_control/handler_impl.cc#L210

I explicitly initialize it in the constructor for safety. Then use it to determine if check was called.

Reply:
I see.  Thanks